File: JavaScript/leetcode75/src/test/Problem36_PathSumIII.test.js
const pathSum = require("../main/Problem36_PathSumIII");

//  Definition for a binary tree node.
function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

test("should return total paths possible", () => {
  let root = new TreeNode(10, 5, -3);
  root.left = new TreeNode(5, 3, 2);
  root.left.left = new TreeNode(3, 3, -2);
  root.right = new TreeNode(-3, null, 11);
  root.left.right = new TreeNode(2, null, 1);
  root.left.right.right = new TreeNode(1, null, null);

  expect(pathSum(root)).toBe(3);
});

File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem36_PathSumIIITest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;
import org.phoenix.leetcode.challenges.Problem33_MaximumDepthOfBinaryTree.TreeNode;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Problem36_PathSumIIITest {

    private final Problem36_PathSumIII test = new Problem36_PathSumIII();

    @Test
    void pathSum() {
        TreeNode root = new TreeNode(10);
        root.left = new TreeNode(5);
        root.left.left = new TreeNode(3);
        root.left.left.left = new TreeNode(3);
        root.left.left.right = new TreeNode(-2);
        root.left.right = new TreeNode(2);
        root.right = new TreeNode(-3);
        root.right.right = new TreeNode(11);
        root.left.right.right = new TreeNode(1);

        assertEquals(3, test.pathSum(root, 8));
    }
}